java-sdk/connectors/security-filtering-connector/src/main/java/com/lucidworks/connector/plugins/security/fetcher/SecurityFilteringAccessControlFetcher.java

```diff
@@ -1,74 +1,76 @@
-        if (!Strings.isNullOrEmpty(assigned)) {
-          context.newAccessControl(input.getId())
-              .metadata(m -> m.setString(SecurityConstants.TYPE, SecurityConstants.PERMISSION_TYPE))
-              .addAllInbound(Collections.singletonList(assigned))
-              .emit();
-        }
-        break;
+    return context.newResult();
+  }
 
-      case SecurityConstants.USER_TYPE:
-        String parentGroup = (String) metadata.getOrDefault(SecurityConstants.PARENTS, null);
-        logger.info("Access-control type={}, metadata={}", type, metadata);
-        List<String> outbound = Collections.emptyList();
-        if (!Strings.isNullOrEmpty(parentGroup)) {
-          outbound = Collections.singletonList(parentGroup);
-        }
-        context.newAccessControl(input.getId())
-            .metadata(m -> m.setString(SecurityConstants.TYPE, SecurityConstants.USER_TYPE))
-            .fields(f -> {
-              f.setString("fullName", input.getId() + "FullName");
-              f.setInteger("internalId", input.getId().hashCode());
-            })
-            .addAllOutbound(outbound)
-            .emit();
-        break;
+  private void createDocsAndAcs(FetchContext context) {
+    context.newGraphAccessControl(USER1).metadata(m -> m.setString("type", "user")).addAllInbound(Collections.singletonList(USER1)).emit();
+    context.newGraphAccessControl(USER2).metadata(m -> m.setString("type", "user")).addAllInbound(Collections.singletonList(USER2)).emit();
+    context.newGraphAccessControl(GROUP1).metadata(m -> m.setString("type", "group")).addAllInbound(Collections.singletonList(USER1)).emit();
+    List<String> users = new ArrayList();
+    users.add(USER1);
+    users.add(USER2);
+    context.newGraphAccessControl(GROUP2).metadata(m -> m.setString("type", "group")).addAllInbound(users).emit();
+    createDocuments(context);
+    logger.info("Fetch created 2 users 2 groups and 2 documents");
+  }
 
-      case SecurityConstants.GROUP_TYPE:
-        context.newAccessControl(input.getId())
-            .metadata(m -> m.setString(SecurityConstants.TYPE, SecurityConstants.GROUP_TYPE))
-            .fields(f -> {
-              f.setString("fullName", input.getId() + "FullName");
-              f.setInteger("internalId", input.getId().hashCode());
-            })
-            .emit();
-        break;
+  private void createDocuments(FetchContext context) {
+    context.newDocument(DOC1).fields(f -> f.setStrings(ACL_FIELD, Collections.singletonList(GROUP1))).emit();
+    context.newDocument(DOC2).fields(f -> f.setStrings(ACL_FIELD, Collections.singletonList(GROUP2))).emit();
+  }
 
-      default:
-        logger.warn("Invalid type: [{}]", type);
-    }
-    return context.newResult();
+  private void updateACs(FetchContext context, String id) {
+    logger.info("Updating {} id {} ", USER1, id);
+    context.newUpdateGraphAccessControlItem(id, "updateValue").emit();
```

Comment:
I kept it out of README because you thought it will not be needed for the SP and Ldap connectors. Since this was
very specific for those connectors, it will be probably be dropped out altogether if not used there. It requires specifying the doc-id for each shard because SP uses a query to find which documents should be updated.